sympy/functions/elementary/trigonometric.py

```diff
@@ -2867,7 +2867,7 @@ class atan2(InverseTrigonometricFunction):
 
     >>> from sympy import atan
     >>> atan2(y, x).rewrite(atan)
-    2*atan(y/(x + sqrt(x**2 + y**2)))
+    Piecewise((2*atan(y/(x + sqrt(x**2 + y**2))), Ne(y, 0)), (pi, x < 0), (0, x > 0), (nan, True))
```

Comment:
Maybe it would be better to return this expression directly from rewrite. It seems to be a simplification of the one currently returned.

Reply:
I'm sorry, I didn't understand that, can you elaborate please?